Gzip + ops

wpears opened the pull request:
Loader now supports gzipping for local files, urls, and files on S3.

Bundled in for fun with this PR:
  - pump for the main pipeline
  - shrinkwrapped dependencies
  - various bug fixes/expanded error handling
  - updated Dockerfile
  - formatting

wpears commented:
Found a latent bug, hold on merge until another commit lands

wpears commented:
good to go :boom:

hkeeler commented on loadtiger.sh:
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+find $1 -type f -iname "tl_2014_$2*.zip" -print0 | while IFS= read -r -d $'\0' file;
+do
+  ./grasshopper-loader.js -d "${file}" -t tiger --index census --type addrfeat -h awsdevhmdal08 -l error
```
You have a hard-coded internal server name here.

wpears replied:
@_@ fixed.

hkeeler commented on grasshopper-loader.js:
```diff
@@ -119,34 +138,39 @@ function run(program, loaderCallback){
+
+      function(err){
+        if(err){
+          console.log('Streaming of %s interrupted by error,', fileName);
```
Seems like it would be useful to have to underlying error part of this message...or is this logged upstream somewhere by returning `loaderCallback(err)`?

wpears replied:
Correct, it's thrown/logged as part of the callback. Lines 62-72 of `grasshopper-loader.js`

hkeeler commented on package.json:
```diff
@@ -27,28 +27,25 @@
   "scripts": {
     "test": "node test/test.js"
   },
-  "devDependencies": {
```
Why get rid of `devDependencies`?  Shouldn't libs like `tape` go here?

wpears replied:
This was due to a perhaps improper use of shrinkwrap. Shrinkwrap doesn't copy over devDependencies (since it is primarily intended for production), yet I was shrinkwrapping everything. Thus, dev installs weren't being filled with needed testing modules. It might make more sense to just drop the shrinkwrap file and return to devDependencies actually being meaningful.

hkeeler commented:
Review complete.  Just a few open questions...

hkeeler commented:
Looks good.  Once Travis goes green, I'll merge.

hkeeler commented:
Travis seems...displeased.

wpears commented:
Some external deps were modified and fixed in #96.. but I didn't update them here so the new build broke on travis. fixed/ patched  so travis is happy. Final updates along these lines are captured in #96.

hkeeler commented:
Looks good.  Merging.
